### ISSUE TITLE
New mutator proposal: remove item from literal array

### ISSUE THREAD

**Issue by MidnightDesign** [infection/infection#597]
TL;DR: A "remove item from literal array" mutator would be great.

**Is your feature request related to a problem? Please describe.**
I'm doing kind of a fingerprinting/checksum thing, like:
```php
$relevantData = [$item->foo(), $item->bar()];
$fingerprint = md5(json_encode($data));
```
I just added a third item to the list of relevant data: `[$item->foo(), $item->bar(), $item->baz()]`. A little later I realized that whenever the output of `baz()` changes, the output of `foo()` changes too. So I could remove the first item and have the (functionally) same thing.

Infection didn't check if the array item was required.

**Describe the solution you'd like**
Infection should warn me that an element in an array literal could be removed with the tests still passing.

**Comment by localheinz** [comment]
I think it is good idea - I believe the challenge here is to decide how many mutations we want to yield from such a mutator.

**Comment by borNfreee** [comment]
Since we are trying to keep balance between the number of generated mutants and the speed of infection, I would say that generating N mutations for an array with N elements is too much.

What about generating 1 mutation _by default_ and have a setting to enable N mutations if someone needs a hardcore mode? :)

**Comment by MidnightDesign** [comment]
Maybe something like "N mutations for an array with N elements, up to X elements. If there are more than X elements, generate only one mutation for the first element". (Where X could be configured.)

**Comment by localheinz** [comment]
Perhaps something like using a `--level` option, similar to `phpstan/phpstan`?

Then it wouldn't be necessary to have a complicated configuration, but something that applies to all mutators that could potentially yield more than one mutation?

Also see #636.
